python/cog/predictor.py

```diff
@@ -46,7 +47,9 @@
 
 
 class BasePredictor(ABC):
-    def setup(self, weights: Optional[Union[CogFile, CogPath]] = None) -> None:
+    def setup(
+        self, weights: Optional[Union[CogFile, CogPath]] = None
+    ) -> Union[Awaitable[None], None]:
```

Comment:
`Union[T, None]` is usually spelled `Optional[T]`, right? Any reason we wouldn't do that here?